packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -715,7 +715,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.blobManager = new BlobManager(
             this.IFluidHandleContext,
             () => this.storage,
-            (blobId) => this.submit(ContainerMessageType.BlobAttach, undefined, undefined, { blobId }),
+            (blobId) => {
+                if (this.attachState === AttachState.Detached) {
+                    this.blobManager.addBlobId(blobId);
```

Comment:
Will the BlobAttach op never be sent in this case? Does attachGraph() get called when the container itself is still detached?

Reply:
When container is detached, we can't send any ops, as they will precede summary for whole container that will be created later in time to create a file. I believe that we attach all data stores (and thus all DDSs, and thus all blob handles that are stored in DDSs) right before container switched to attaching mode.
Take a look at setAttachState() & "attaching" events.

That said, I see that Container raises "attaching" event after grabbing summary, not before it.
@jatgarg, can you please clarify how it works? I.e. If I put a handle to a DDS, and this DDS is reachable from conatiner, when such handle would get attached in the process of attaching container?

@wes-carlson - one thing I realized that code I've added will not work. That's because there is no way to upload a blob to strorage if we have no file. So adding images in detached state is a feature that needs to be built. I'll add asserts to ensure whoever hits it understand what needs to be done, I'm not going to build it in this PR :)